packages/jest-runtime/src/index.ts

```diff
@@ -833,6 +843,30 @@ class Runtime {
     this._moduleMocker.clearAllMocks();
   }
 
+  teardown(): void {
```

Comment:
`_explicitShouldMock` is not cleared here

Reply:
it's cleared in `resetModules()` which is now called (wasn't when you reviewed, though, #9906)

Reply:
Um, can you point me there? I don't see any reference to clearing it, only setting/getting

Reply:
No you're right. Not sure how I missed it... Probably not super safe to clear in `resetModules` regardless, so I'll stick it in here

Reply:
eb5e4ab